homeassistant/components/media_player/squeezebox.py

```diff
@@ -80,17 +80,12 @@ def __init__(self, host, port, username, password):
     def _get_http_port(self):
         """Get http port from media server, it is used to get cover art."""
         http_port = None
-        try:
-            http_port = self.query('pref', 'httpport', '?')
-            if not http_port:
-                _LOGGER.error("Unable to read data from server %s:%s",
-                              self.host, self.port)
-                return
+        http_port = self.query('pref', 'httpport', '?')
+        if http_port:
```

Comment:
Wrote that comment pre-coffee so my apologies that it was a bit unclear. What I was looking for was not setting `http_port` twice. If the `self.query` fails does it return `None` or something else? I'd be happy with this code if it returns None:

```python
http_port = self.query('pref', 'httpport', '?')
if http_port is None:
    _LOGGER.error("Failed to connect to server %s:%s",
                  self.host, self.port)
return http_port
```